### PR TITLE
#167190523 Setup Unit Test

### PR DESCRIPTION
**Pull request by Adekoreday**
#### What does this PR do?
This PR setup unit test using JEST and super test

#### Description of Task proposed in this pull request?
The app server file was split into two,
 one for listening on the default port 
the other for contains the remaining server files

reason for this is to prevent jest from listening to the PORT after test.

#### How should this be manually tested (Quality Assurance)?

Run the following commands
```

npm  test

```
#### What are the relevant pivotal tracker stories?
 [#167190523](https://www.pivotaltracker.com/story/show/167190523)

#### Any background context you want to add 
```
npm install 
```

To test, the user must reinstall updated packages 
        
All features would need to include a unit test moving forward.

#### What I have learned working on this feature
I learnt how to set up Jest for node-express server-side application

#### Screenshots: 

Screenshot 2019-07-23 at 11 50 35 PM

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,3 +1,12 @@
 {
-     "presets": ["@babel/preset-env"]
+     "presets": [
```
Comment (dinobi) [inline .babelrc]: how many line spacing are we adhering to?. I'll advice you setup a `.editorconfig` to help with this. Make sure the team uses that

Comment (Adekoreday) [reply .babelrc]: thanks will adhere to this and inform the team moving forward

File: __test__/index.test.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+
+import request from 'supertest';
+import app from '../server';
+
+
+let server;
+let agent;
+beforeEach((done) => {
```
Comment (dinobi) [inline __test__/index.test.js]: Can we make this a beforeAll?

File: __test__/index.test.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  server = app.listen(4000, (err) => {
+    if (err) return done(err);
+    agent = request.agent(server);
+
+    done();
+  });
+});
+
+afterEach(done => server && server.close(done));
```
Comment (dinobi) [inline __test__/index.test.js]: `afterAll` here

File: __test__/index.test.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+
+import request from 'supertest';
+import app from '../server';
+
+
+let server;
+let agent;
```
Comment (dinobi) [inline __test__/index.test.js]: what if we rename this to `http`, so it reads nicely with the test `await http.get('/')`

File: __test__/index.test.js (hunk 2 above)
Comment (Adekoreday) [reply __test__/index.test.js]: okay, this is to reduce the no of starting and stopping the server on each test case

File: __test__/index.test.js (hunk 4 above)
Comment (Adekoreday) [reply __test__/index.test.js]: okay, will do that

File: __test__/index.test.js (hunk 3 above)
Comment (Adekoreday) [reply __test__/index.test.js]: done

**Review by dinobi — APPROVED** [review]

**Review by abdulfatai360 — APPROVED** [review]
Good job, Adekorede!

**Review by JuwonAbiola — APPROVED** [review]

**Review by chialuka — APPROVED** [review]

**Pull request merged**